src/main/java/Heap.java

```diff
@@ -60,6 +149,41 @@ boolean isEmpty() {
      * @return Number of elements within the heap
      */
     int size() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("[");
+        for (int i = 0; i < size(); i++) {
+            stringBuilder.append(heap[i]);
+            if (i < size() - 1) {
+                stringBuilder.append(", ");
+            }
+        }
+        stringBuilder.append("]");
+        return stringBuilder.toString();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Heap<?> that = (Heap<?>) o;
+        return Arrays.equals(this.heap, 0, this.size(), that.heap, 0, that.size());
```

Comment:
IIRC heaps are not unique, in that you'd be able to construct two heaps that always behave identically in all their operations but the array content order is not equal.

Reply:
Upon reading, you call this out in the description. I did say _My brain does not want to do things_.

Though, given that, it may make more sense to define this equality locally in the tests rather than in the class.  
When you get `false` back from this equality test, you don't actually know if they were not equal. Further, the comparator is not used in the equality test at all, so when you get `true` you don't actually know if they were equal either.  
Defining it locally in the test class lets you brush away the inconsistencies without any major consequences.

Reply:
For the sake of keeping the equals in the class, I'm thinking

1. Make a copy of the heaps
2. Make it check equality on the remove order
3. Check comparator equality too

Reply:
Seems reasonable, but the comparator will be a little finicky. Two comparators are only equal if they are the same object. Given the equality test will be hidden in the test, I don't think it's a big deal, just something to remember.